cmds/readlink/readlink.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// readlink display value of symbolic link file
+//
+// Synopsis:
+//     readlink [OPTIONS] FILE
+//
+// Options:
+//     -f: follow
+//     -v: verbose
+//
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+var (
+	follow  = flag.Bool("f", false, "follow recursively")
+	verbose = flag.Bool("v", false, "report error messages")
+)
+
+func isLink(file string) error {
+	var err error
+	f, err := os.Lstat(file)
+
+	if err != nil {
+		return err
+	}
+
+	// If it is not a symbolic link return an error
+	if f.Mode()&os.ModeSymlink == 0 {
+		return fmt.Errorf("%s Invalid argument", file)
+	}
+
+	return err
+}
+
+func readFile(file string) error {
+	var path string
+	var err error
+
+	err = isLink(file)
```

Comment:
super nit: it's customary to embed this in the if-statement, especially if nothing else is declared:

```go
if err := isLink(file); err != nil {
  return err
}
```